packages/node_modules/@webex/internal-plugin-calendar/test/unit/spec/calendar.js

```diff
@@ -0,0 +1,138 @@
+/*!
+ * Copyright (c) 2015-2019 Cisco Systems, Inc. See LICENSE file.
+ */
+
+import {assert} from '@webex/test-helper-chai';
+import Calendar from '@webex/internal-plugin-calendar';
+import MockWebex from '@webex/test-helper-mock-webex';
+import sinon from 'sinon';
+
+import {CALENDAR_REGISTERED, CALENDAR_UPDATED, CALENDAR_DELETE, CALENDAR_CREATE, CALENDAR_UNREGISTERED} from '../../../src/constants';
+
+describe('internal-plugin-calendar', () => {
+  describe('Calendar Apis', () => {
+    const webex = new MockWebex({
+      children: {
+        calendar: Calendar
+      }
+    });
+
+    beforeEach(async () => {
+      webex.canAuthorize = true;
+      webex.internal.device = {
+        register: sinon.stub().returns(Promise.resolve()),
+        unregister: sinon.stub().returns(Promise.resolve())
+      };
+      webex.internal.mercury = {
+        connect: sinon.stub().returns(Promise.resolve()),
+        disconnect: sinon.stub().returns(Promise.resolve()),
+        on: sinon.stub().callsFake((event, callback) => {
+          if (event === 'event:calendar.meeting.update') {
+            callback({data: {calendarMeetingExternal: {id: 'calendarId1'}}});
+          }
+          if (event === 'event:calendar.meeting.delete') {
+            callback({data: {calendarMeetingExternal: {id: 'calendarId1'}}});
+          }
+          if (event === 'event:calendar.meeting.create') {
+            callback({data: {calendarMeetingExternal: {id: 'calendarId1'}}});
+          }
+        }),
+        off: sinon.spy()
+      };
+      await webex.internal.calendar.unregister();
+
+      webex.request = sinon.stub().returns(Promise.resolve({body: {items: [{id: 'calendar1'}, {id: 'calendar2'}]}}));
```

Comment:
Is this mock needed for all the tests or just certain ones? Can we move it out if it isn't needed in all of the tests?